### PR TITLE
Gith

### PR DESCRIPTION
**Pull request by farhanlion**
The GitHub sign-in is complete.

### CODE REVIEW

File: app/controllers/users/omniauth_callbacks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+require "open-uri"
+require "json"
+
+class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  def github
+    user = User.find_for_github_oauth(request.env['omniauth.auth'])
+    if user == false
+      session['devise.github_data'] = request.env['omniauth.auth']
+      set_flash_message(:alert, :not_a_wagon_student, kind: 'Github')
```
Comment (mirhamasala) [inline app/controllers/users/omniauth_callbacks_controller.rb]: What is this `:not_a_wagon_student` about? Do we need that?

File: app/controllers/users/omniauth_callbacks_controller.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      redirect_to new_user_registration_url
+    elsif user.persisted?
+      set_flash_message(:notice, :success, kind: 'Github') if is_navigational_format?
+      sign_in_and_redirect user, event: :authentication
+      # user.first_login ? redirect_to(edit_user_registration_path) : redirect_to(root_path)
```
Comment (mirhamasala) [inline app/controllers/users/omniauth_callbacks_controller.rb]: Commented out code can be removed if you don't plan to use it anymore. 🙏🏽

File: app/models/user.rb [hunk 3]
```diff
@@ -2,12 +2,55 @@ class User < ApplicationRecord
...
+      user = User.new(user_params)
+      user.password = Devise.friendly_token[0,20]  # Fake password for validation
+      user.save
+    else
+      return false
+    end
+
+    return user
+  end
+
```
Comment (mirhamasala) [inline app/models/user.rb]: Do we need to identify if users are Wagon students?

File: app/views/devise/sessions/new.html.erb [hunk 4]
```diff
@@ -22,6 +22,15 @@
...
   </div>
 
+  <div>
+    <%- if devise_mapping.omniauthable? %>
+  <%- resource_class.omniauth_providers.each do |provider| %>
+    <%= link_to "Sign in with #{OmniAuth::Utils.camelize(provider)}", omniauth_authorize_path(resource_name, provider) %><br />
+  <% end %>
+<% end %>
+    
+  </div>
```
Comment (mirhamasala) [inline app/views/devise/sessions/new.html.erb]: Please check indentation!

File: app/views/shared/_navbar.html.erb [hunk 5]
```diff
@@ -17,7 +17,9 @@
           </li>
         <% end %>
         <li class="nav-item dropdown">
-          <%= image_tag "", class: "avatar dropdown-toggle", id: "navbarDropdown", data: { toggle: "dropdown", handle: current_user.github_handle }, 'aria-haspopup': true, 'aria-expanded': false %>
+          <% avatar_url = current_user.github_picture_url || "http://placehold.it/30x30" %>
+          <%= image_tag avatar_url, class: "avatar dropdown-toggle", id: "navbar-wagon-menu", "data-toggle" => "dropdown" %>
+          <%= image_tag "https://kitt.lewagon.com/placeholder/users/ssaunier", class: "avatar dropdown-toggle", id: "navbarDropdown", data: { toggle: "dropdown" }, 'aria-haspopup': true, 'aria-expanded': false %>
```
Comment (mirhamasala) [inline app/views/shared/_navbar.html.erb]: Can you please explain the difference between line 21 + 22.

Comment (farhanlion) [reply app/views/shared/_navbar.html.erb]: I removed line 22.

File: app/models/user.rb (hunk 3 above)
Comment (farhanlion) [reply app/models/user.rb]: No we do not. I am removing it.

File: app/controllers/users/omniauth_callbacks_controller.rb (hunk 1 above)
Comment (farhanlion) [reply app/controllers/users/omniauth_callbacks_controller.rb]: No. I removed it.

File: app/views/devise/sessions/new.html.erb (hunk 4 above)
Comment (farhanlion) [reply app/views/devise/sessions/new.html.erb]: I fixed it.

**Pull request merged**